src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -1487,6 +1485,42 @@ class OptionsSegment(BaseSegment):
     )
 
 
+# ReferenceDefinitionGrammar
+class ForeignKeyReferenceSegment(BaseSegment):
+    """A foreign key reference segment."""
+
+    type = "foreign_key_reference_segment"
+    match_grammar = Sequence(
+        "REFERENCES",
+        Ref("TableReferenceSegment"),
+        Ref("BracketedColumnReferenceListGrammar"),
+        "NOT",
+        "ENFORCED",
+    )
```

Comment:
Actually now I'm wondering if it's weird that the Foreign key has a special segment type and the primary key doesn't when it's only slightly simpler.

Given the other dialects don't pull this out into it's own class I think we should remove this, and just add the grammar to the `TableConstraintSegment`.

Reply:
But its an almost completely different syntax - to my eye at least.

Reply:
I don't mean combine them into one section for both. I mean like this:

https://github.com/sqlfluff/sqlfluff/blob/ee8e5365a223ac1047462345d4b4eed816467120/src/sqlfluff/dialects/dialect_ansi.py#L3068-L3089

Reply:
So I inlined it.
I looked at using `Ref(ReferenceDefinitionGrammar)` but the definition is different.